src/main/java/com/google/sps/DataServlet.java

```diff
@@ -153,8 +162,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (mutationNumber > currDataGraph.numMutations()) {
       diff = Utility.getMultiMutationAtIndex(mutList, mutationNumber);
     }
+    List<String> nodeNames = new ArrayList<>();
+    try {
+      JsonParser jsonParser = new JsonParser();
```

Comment:
I realized that when I was writing this I was using older documentation and this constructor is now deprecated. Could you replace these two lines with a call to [`JsonParser.parseString()`](https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/JsonParser.html#parseString(java.lang.String)). The new method also throws a different exception so we'll have to check for that too.